phantomjs-env/src/main/scala/org/scalajs/jsenv/phantomjs/RetryingComJSEnv.scala

```diff
@@ -30,165 +30,103 @@ import scala.util.{Try, Failure, Success}
  *  (e.g. writing to a file). It is the users responsibility to ensure this
  *  property.
  *
- *  No retrying is performed for synchronous, or normal asynchronous runs.
+ *  No retrying is performed for non-com runs (initiated with `start`).
  *
  *  Although `RetryingComJSEnv` is agnostic of the underlying JS env, and is
  *  therefore not tied to PhantomJS, it is most often used to compensate for
  *  flakiness effects of PhantomJS.
  */
-final class RetryingComJSEnv(val baseEnv: ComJSEnv,
-    val maxRetries: Int) extends ComJSEnv {
+final class RetryingComJSEnv(val baseEnv: JSEnv, val maxRetries: Int)
+    extends JSEnv {
 
-  def this(baseEnv: ComJSEnv) = this(baseEnv, 5)
+  def this(baseEnv: JSEnv) = this(baseEnv, 5)
 
-  def name: String = s"Retrying ${baseEnv.name}"
+  val name: String = s"Retrying ${baseEnv.name}"
 
-  def jsRunner(files: Seq[VirtualJSFile]): JSRunner =
-    baseEnv.jsRunner(files)
+  def start(input: Input, runConfig: RunConfig): JSRun =
+    baseEnv.start(input, runConfig)
 
-  def asyncRunner(files: Seq[VirtualJSFile]): AsyncJSRunner =
-    baseEnv.asyncRunner(files)
-
-  def comRunner(files: Seq[VirtualJSFile]): ComJSRunner =
-    new RetryingComJSRunner(files)
-
-  /** Hack to work around abstract override in ComJSRunner */
-  private trait DummyJSRunner {
-    def stop(): Unit = ()
+  def startWithCom(input: Input, runConfig: RunConfig,
+      onMessage: String => Unit): JSComRun = {
+    new RetryingJSComRun(input, runConfig, onMessage)
```

Comment:
This doesn't handle non-inherited I/O properly. TBH I think we should just drop this entire env. It is way too over engineered for what it is useful for.

Reply:
#19